utils/SuppressionsManager.py

```diff
@@ -0,0 +1,125 @@
+import os
+import json
+from utils.Tools import _warn, _info
+
+class SuppressionsManager:
+    """
+    Manages suppressions for Service Screener findings.
+    Loads suppressions from a JSON file and provides methods to check if a finding should be suppressed.
+    """
+    
+    def __init__(self):
+        self.suppressions = {}
+        self.is_loaded = False
+        self.suppression_count = 0
+    
+    def __getstate__(self):
+        """Support for pickling (multiprocessing compatibility)."""
+        return self.__dict__
+    
+    def __setstate__(self, state):
+        """Support for unpickling (multiprocessing compatibility)."""
+        self.__dict__.update(state)
+        
+    def load_suppressions(self, file_path):
+        """
+        Load suppressions from a JSON file.
+        
+        Args:
+            file_path (str): Path to the JSON suppressions file
+            
+        Returns:
+            bool: True if suppressions were loaded successfully, False otherwise
+        """
+        if not file_path or not os.path.exists(file_path):
+            _warn(f"Suppression file not found: {file_path}")
+            return False
+            
+        try:
+            with open(file_path, 'r') as f:
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Path traversal vulnerability detected. User-controlled input in file paths can allow attackers to access files outside intended directories using `../` sequences. Secure your code by using framework functions like `safe_join()`, `secure_filename()` and `.startswith()` checks. Learn more: https://cwe.mitre.org/data/definitions/22.html